WordPress/Classes/Stores/StatsWidgetsStore.swift

```diff
@@ -258,22 +258,21 @@ private extension StatsWidgetsStore {
     /// Observes WPAccountDefaultWordPressComAccountChanged notification and reloads widget data based on the state of account.
     /// The site data is not yet loaded after this notification and widget data cannot be cached for newly signed in account.
     func observeAccountChangesForWidgets() {
-        NotificationCenter.default.addObserver(forName: .WPAccountDefaultWordPressComAccountChanged,
-                                               object: nil,
-                                               queue: nil) { notification in
+        NotificationCenter.default.addObserver(self, selector: #selector(handleAccountChangedNotification), name: .WPAccountDefaultWordPressComAccountChanged, object: nil)
+    }
 
-            UserDefaults(suiteName: WPAppGroupName)?.setValue(AccountHelper.isLoggedIn, forKey: AppConfiguration.Widget.Stats.userDefaultsLoggedInKey)
+    @objc func handleAccountChangedNotification() {
+        UserDefaults(suiteName: WPAppGroupName)?.setValue(AccountHelper.isLoggedIn, forKey: AppConfiguration.Widget.Stats.userDefaultsLoggedInKey)
 
-            if !AccountHelper.isLoggedIn {
-                HomeWidgetTodayData.delete()
-                HomeWidgetThisWeekData.delete()
-                HomeWidgetAllTimeData.delete()
+        if !AccountHelper.isLoggedIn {
+            HomeWidgetTodayData.delete()
+            HomeWidgetThisWeekData.delete()
+            HomeWidgetAllTimeData.delete()
 
-                UserDefaults(suiteName: WPAppGroupName)?.setValue(nil, forKey: AppConfiguration.Widget.Stats.userDefaultsSiteIdKey)
-                WidgetCenter.shared.reloadTimelines(ofKind: AppConfiguration.Widget.Stats.todayKind)
-                WidgetCenter.shared.reloadTimelines(ofKind: AppConfiguration.Widget.Stats.thisWeekKind)
-                WidgetCenter.shared.reloadTimelines(ofKind: AppConfiguration.Widget.Stats.allTimeKind)
-            }
+            UserDefaults(suiteName: WPAppGroupName)?.setValue(nil, forKey: AppConfiguration.Widget.Stats.userDefaultsSiteIdKey)
```

Comment:
One could DRY and this a tiny bit by storing the `UserDefatuls` instance and adding an early return

```swift
let defaults = UserDefaults(suiteName: WPAppGroupName)

defaults.setValue...

guard !AccountHelper.isLoggedIn else { return }

...
```

🤷‍♂️

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20995/commits/66ff6fb6eca37b11b83c97f4231844a7276a9cad